cookbooks/cdo-varnish/test/integration/cloudfront/minitest/test_default.rb

```diff
@@ -0,0 +1,315 @@
+# Chef test suite for CloudFront distribution behavior.
+
+# Chef node attributes as constants
+DASHBOARD_PORT = 8080
+PEGASUS_PORT = 8081
+VARNISH_PORT = 80
+LOCALHOST = 'localhost'
+VARNISH_URL = "#{LOCALHOST}:#{VARNISH_PORT}"
+ENVIRONMENT = 'integration'
+CLOUDFRONT_URL = "https://#{ENVIRONMENT}-studio.code.org"
+
+# Setup the mock-server and ngrok
+puts 'setup'
+NGROK_PID = spawn('/usr/local/ngrok/ngrok start cdo --config=/home/kitchen/.ngrok2/ngrok.yml --log=stdout')
+PID = spawn('cd ~; java -jar mock.jar')
+# Don't start tests until wiremock is live.
+mock_started = false
+until mock_started
+  `sleep 1`
+  puts 'testing mock'
+  `curl -s -i #{LOCALHOST}:#{DASHBOARD_PORT}/start`
+  mock_started = $?.exitstatus == 0
+end
+puts 'setup finished'
+
+at_exit do
+  puts 'teardown'
+  `curl -s -X POST #{LOCALHOST}:#{DASHBOARD_PORT}/__admin/shutdown`
+  Process.wait PID
+  `kill -INT #{NGROK_PID}`
+  Process.wait NGROK_PID
+  puts 'teardown finished'
+end
+
+require 'json'
+require 'minitest/autorun'
+
+describe 'varnish' do
+  it 'is installed' do
+    assert_equal '/usr/sbin/varnishd', `which varnishd`.strip
+  end
+
+  it 'is running' do
+    assert_equal 0, (`service varnish status` && $?.exitstatus)
+  end
+end
+
+# Returns the number of times the specified stub-URL was accessed.
+def count_requests(url)
+  request = <<JSON
+{
+  "method": "GET",
+  "url": "#{url}"
+}
+JSON
+  JSON.parse(`curl -s -X POST #{LOCALHOST}:#{DASHBOARD_PORT}/__admin/requests/count -d '#{request}'`)['count']
+end
+
+# Mocks a simple text/plain response body at the specified URL.
+def mock_url(url, body, request_headers={}, response_headers={}, method='GET')
+  json = {
+  request: {
+    method: method,
+    url: url,
+    headers: Hash[*request_headers.map{|k, v|[k,{equalTo: v}]}.flatten]
+  },
+  response: {
+    status: 200,
+    body: body,
+    headers: {
+      'Cache-Control' => 'public, max-age=10',
+      'Content-Type' => 'text/plain'
+    }.merge(response_headers)
+  }
+}.to_json
+  `curl -s -X POST #{LOCALHOST}:#{DASHBOARD_PORT}/__admin/mappings/new -d '#{json}'`
+end
+
+def _request(url, headers={}, cookies={}, method='GET')
+  header_string = headers.map { |key, value| "-H \"#{key}: #{value}\"" }.join(' ')
+  cookie_string = cookies.empty? ? '' : "--cookie \"#{cookies.map{|k,v|"#{k}=#{v}"}.join(';')}\""
+  `curl -X #{method} -s #{cookie_string} #{header_string} -i #{url}`.tap{assert_equal 0, $?.exitstatus}
+end
+
+def request(url, headers={}, cookies={})
```

Comment:
Could you call this "dashboard_request" or something like that to clarify that this is a request to the dashboard.

Then you could also rename "_request" to simply "request".  It looked like an internal method based on the naming convention but then I saw that it was being called directly in the tests below.